pkg/storage/stores/indexshipper/compactor/deletion/delete_requests_manager.go

```diff
@@ -149,6 +152,11 @@ func (d *DeleteRequestsManager) loadDeleteRequestsToProcess() error {
 		if deleteRequest.EndTime > ur.requestsInterval.End {
 			ur.requestsInterval.End = deleteRequest.EndTime
 		}
+
+		deleteCount++
+		if deleteCount >= d.batchSize {
+			break
```

Comment:
so when we go over batch size, we just bail and go on to the next request? what happens to the requests that didn't get processed? are they just dropped, queued up for the next delete?

In any case I think a log message might be helpful here, to let the user know maybe how many were dropped, or just that the limit was hit, but there are still requests in the queue being skipped?

Reply:
The requests are pulled out by status `Received`. When they're completed, the status is updated to processed. This change means that the remaining deletes wait to be processed (but filtering continues).